tests/test_application.py

```diff
@@ -1680,3 +1680,28 @@ async def raise_method(*args, **kwargs):
                 app.run_webhook(close_loop=False, stop_signals=None)
 
         assert len(recwarn) == 0
+
+    @pytest.mark.dev
+    def test_signal_handlers(self, app, monkeypatch):
+        # this test should make sure that signal handlers are set by default on Linux + Mac,
+        # and not on Windows.
+
+        def signal_handler_test(*args, **kwargs):
+            # args[0] is the signal, [1] the callback
+            assert args[0] in (signal.SIGINT, signal.SIGTERM, signal.SIGABRT)
```

Comment:
This doesn't test anything on windows, because it's never called. suggestion: something like `received_signals.append(args[0])` and then later on

```
if windows:
    assert received_signals == []
else:
    assert received_signals = [signal.SIGINT, signal.SIGTERM, signal.SIGABRT]
```

Reply:
I am fine with that, this is supposed to test the default. If the raise test later fails at one CI run, we know we can switch the defaults back.

Reply:
the problem is that it wouldn't fail if we pass a signal on windows because the signal would still be one of the ones that you specified here.

Reply:
no but then it would raise the error, as it does below...

Reply:
the current CI failures are in `test_run_stop_signal_warning_windows` which expects a warning to be issues on windows and has to be adapted. In the current state doing `run(stop_signal=(Signal.SIGINT,))` would not raise an exception because you mocked the method that raises `NotImplementedError`.

Reply:
> It does not though? It expects a totally different warning? And it gets it, but for some reason expects it to not be there?

Fair enough: It expects `stop_signal=None` *not* to issue a warning, but it still does due to https://github.com/python-telegram-bot/python-telegram-bot/pull/3065#discussion_r879858634.

Reply:
this is still open :) i.e.

* `test_run_stop_signal_warning_windows` needs to be adapted and 
* IMO it would still be good to explicitly check that we don't pass signals on windows